### PR TITLE
Talent Search - Add SupportForm component

### LINKED ISSUE #4188: Build Support Form in Storybook
Build the form shown on the FAQ page designs: [figma prototype](https://www.figma.com/file/NhEMp9tiBNi5VwuhmRr4JW/Browse-jobs%2C-Support-and-FAQs?node-id=11%3A721)

Note: the page is still in flux, so for now just build a simple functional version of the form with minimal styling.

The payload should designed to submit to the [freshdesk api](https://developers.freshdesk.com/api/#create_ticket). It must include only the basic fields discussed in #4127:
- Name input: name
- Email input: email
- Subject dropdown: type
    - the option values should be strings, eg bug, feedback, and question
- Details textarea: description

## Acceptance Criteria

- [x] If user is logged in, prepopulate Name and Email
- [x] When submitted, sends a POST request (using fetch API) to an endpoint defined in a build-time environment variable. If env variable is undefined or empty, throw an error.
- [x] Attaches an API key also defined in a build-time env variable
- [x] If response includes messages in the `error` key, display them in error toast?
- [x] If submission succeeds: replace the form with text saying that ticket submission was successful, with a button to show the form again

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #4188.

### Steps to test
1. Launch talentsearch storybook `npm run storybook --workspace=talentsearch`
2. Search for **Support Form**

### CODE REVIEW

**Comment by vd1992** [comment]
little hard to really judge things, since the styling is tentative and API functionality pending
it appears to work, the request body fills out and the error toast appears 

image

**Comment by mnigh** [comment]
> little hard to really judge things

Fair point, not easy to test the request and response in storybook. You could create a test account in Freshdesk and add the `env` variables, not ideal, but we don't have access to a shared account yet (see #4193).

File: frontend/talentsearch/src/js/components/support/SupportForm.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+          defaultMessage:
+            "In the meantime, feel free to check out our FAQs for further information.",
+          id: "QX1l/C",
+          description: "Support form success paragraph three",
+        })}
+      </p>
+      <div data-h2-text-align="base(center)">
+        <Button
+          onClick={() => {
+            onFormToggle(true);
```
Comment (petertgiles) [inline frontend/talentsearch/src/js/components/support/SupportForm.tsx:85]: I wonder if the form should clear, at least the last two fields, on reset?

**Review by petertgiles — APPROVED** [review]
Yeah, really hard to properly test this without a backend set up for it.  The code and story look nice, though.  I guess any bugs will need to get ironed out in later issues.

File: frontend/talentsearch/src/js/components/support/SupportForm.tsx (hunk 1 above)
Comment (petertgiles) [reply frontend/talentsearch/src/js/components/support/SupportForm.tsx:85]: That's fair. :+1:

**Pull request merged**